nifti2/nifti_tool.c

```diff
@@ -3894,7 +3895,12 @@ int modify_field(void * basep, field_s * field, const char * data)
                   return 1;
                }
                /* otherwise, we're good */
-               ((short *)((char *)basep + field->offset))[fc] = (short)val;
+               int16_t sval = (int16_t)val;
+               char * offsetp = (char *)basep + field->offset;
+               offsetp = &(offsetp[fc * sizeof(sval)]);
+               assert(offsetp == &(((short *)((char *)basep + field->offset))[fc])); // TEMP
+               assert(0);
```

Comment:
Can this assert(0) be removed?

Reply:
Are the other assert()s going to stay?  If so, that means some compiler warns about the original statements but not the asserts?
If the asserts will be kept, they should probably have casts to (char *)before the '&'.

Reply:
Also, going from a pointer to an array and back seems a little odd.  What if 3900 were to read:
   `offsetp += fc*sizeof(sval);  /* offset further by fc elements */`

Or that could be merged into the previous line, though it is nice to spell out the pieces:
```
    /* from the struct base offset by the field location, access element #fc */
    char * offsetp = (char *)basep + field->offset + fc*sizeof(sval);
```

Reply:
Please see my comments way above, from 2023-04-01, repeated here:

"Don't merge as-is, I added some asserts to verify the address calculation is correct. They are just there as a test and will be removed.

Alas, no test actually enters this code, as I confirmed with the assert(0)."

It would be good if we first added some tests that exercise this code. Then we can confirm the code change is correct by not hitting those asserts.